src/arti/tor-dirmgr/src/authority.rs

```diff
@@ -44,6 +46,29 @@ impl Authority {
     }
 }
 
+impl FromStr for Authority {
+    type Err = Error;
+
+    /// Parse Authority from a string.
+    fn from_str(authority_raw: &str) -> Result<Self, Self::Err> {
+        // name, v3ident_raw
+        let authority: Vec<&str> = authority_raw.split_whitespace().collect();
+
+        if authority.len() != 2 {
+            bail!("Invalid format for authority.");
```

Comment:
Could be useful to add more info to the error context, e.g. "expected 2 elements, found {}"

Reply:
This password is only used for tests, once the test is done, the authority is destroyed with the Docker container, and we forget about it.

We can add more info, but that would require more string formatting.